src/utils.js

```diff
@@ -38,6 +43,44 @@ function _normalizeUrl(src, baseUrl) {
 }
 module.exports._normalizeUrl = _normalizeUrl;
 
+async function _fetch(url) {
+  const res = await fetch(url)
+  if (res.status >= 200 && res.status < 300) {
+    return await res.text();
+  } else {
+    throw new Error('fetch got invalid status code: ' + res.status + ' : ' + url);
+  }
+}
+module.exports._fetch = _fetch;
+
+var _G = typeof global !== 'undefined' ? global : 
+         typeof window !== 'undefined' ? window :
+         typeof self !== 'undefined' ? self : this;
+var _context = vm.createContext(_G);
```

Comment:
Is this necessary? As far as I remember any context creation invokes a serious performance penalty to proxy across the boundary.

Reply:
It's the only way to match `vm.runInThisContext`. There's no `vm.getGlobalContext` function, so I had to make one myself.

We can turn it off, but then scripts won't run in `ThisContext`.

Reply:
What's the alternative to share the context without a proxy layer?

Reply:
I don't think there is an alternative, really. Either it's running in the global context or it's not.

I might be mistaken about that.

Reply:
I'm certain there's a way, even if it requires editing `node`. Maybe it's possible to run the module in the global context without any contextification? That's what `vm.runInThisContext` does.

Reply:
I looked into the node source; it looks like the context is [a wrap over `v8::Context`](https://github.com/nodejs/node/blob/8772da6ea890afbe9088b71edf98fd3023090b00/src/module_wrap.cc#L131), so we could reasonably get the "current context" into that wrap and pass it in, without any changes to `node`.

Reply:
Actually, looking through the node code it looks like just having no context is fine: https://github.com/nodejs/node/blob/8772da6ea890afbe9088b71edf98fd3023090b00/lib/internal/vm/source_text_module.js#L88

We would have to consider what that means for `type="module"` though.